botcity/core/bot.py

```diff
@@ -494,10 +495,12 @@ def get_element_coords(self, label, x=None, y=None, width=None, height=None, mat
             print('Warning: Ignoring best=False for now. It will be supported in the future.')
 
         ele = pyautogui.locateOnScreen(self._search_image_file(label), region=region, confidence=matching)
-        if is_retina():
-            ele = ele._replace(left=ele.left / 2.0, top=ele.top / 2.0)
-        self.state.element = ele
-        return ele.left, ele.top
+        if ele is not None:
+            if is_retina():
+                ele = ele._replace(left=ele.left / 2.0, top=ele.top / 2.0)
+            self.state.element = ele
+            return ele.left, ele.top
+        return None, None
```

Comment:
Change for a more direct logic instead of nested ifs.

```suggestion
        if ele is None:
            return None, None
        if is_retina():
            ele = ele._replace(left=ele.left / 2.0, top=ele.top / 2.0)
        self.state.element = ele
        return ele.left, ele.top
```